### ISSUE TITLE
Hi，I have the problem in quantized in your paper.

### ISSUE THREAD

**Issue by deeptoby** [tidsp/caffe-jacinto-models#13]
`
class Quantized:
    def __init__(self):
        pass

    @staticmethod
    def check_valid(key):
        invalid = ['running_mean', 'running_var', 'num_batches_tracked']
        for iv in invalid:
            if iv in key:
                return False
        
        return True
    
    def get_range(self, metric):
        minr = float('inf')
        maxr = -float('inf')

        minr = min(minr, torch.min(metric.cpu()).item())
        maxr = max(maxr, torch.max(metric.cpu()).item())
        return minr, maxr     

    @staticmethod
    def clip(metric, mq):
        metric = torch.clamp(torch.round(metric * mq), -128, 127)
        return metric / mq

    def __call__(self, state_dict):
        for key, value in enumerate(state_dict):
            if Quantized.check_valid(value):
                metric = state_dict[value]
                minr, maxr = self.get_range(metric)
                int_len = math.log2(max(abs(minr), abs(maxr))) + 1
                fac_len = 8 - int_len
                
                mq = math.pow(2, fac_len)
                metric = Quantized.clip(metric, mq)
                state_dict[value] = metric
                print("minr: {} maxr: {}".format(minr, maxr))
        print("quantized complete ^_^")
        return state_dict
`
I write it in pytorch, but i have a poor result. I need your help.
1.How can i get the range [Rmin, Rmax]?

**Comment by mathmanu** [comment]
Estimating the ranges is key to doing the quantization. 

For weight quantization, you can directly measure the range of weights as you have the weights available. For activation quantization, it has to be done by actually looking at the ranges of activation. Since you want to use PyTorch, I suggest you to use a code base that implements quantization in PyTorch. I don't have a confident suggestion, but may be you can try https://github.com/NervanaSystems/distiller
